velox/functions/lib/Re2Functions.cpp

```diff
@@ -116,6 +116,60 @@ bool re2Extract(
   }
 }
 
+void checkEscape(bool condition) {
+  VELOX_CHECK(
```

Comment:
This error should not be thrown, but rather reported via context->setError(). See https://facebookincubator.github.io/velox/develop/scalar-functions.html#try-expression-support

Reply:
This happens during the function construction and I think its fine for the error to be thrown there. 

I tried this in Presto
presto:tiny> select * from customer where name like 'C#}#+' escape '#' limit 2;
Query 20211007_210010_00005_m7qj8 failed: Escape character must be followed by '%', '_' or the escape character itself

So the current code seems alright.

wdyt ?